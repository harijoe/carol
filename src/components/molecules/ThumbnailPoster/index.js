import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { theme, breakpoint } from 'utils/style'

import { Heading, Image } from 'components'

const Wrapper = styled.figure`
  position: relative;
  display: flex;
  align-items: center;
  padding: ${theme('spaces.s')};
  height: 15rem;
  overflow: hidden;

  &::before {
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    background: ${theme('colors.grayscale.darkest')};
    opacity: 0.3;
    content: '';
    z-index: 1;
  }

  ${breakpoint('m')`
    height: 22rem;
  `}
`

const StyledImage = styled(Image)`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  min-width: 100%;
  min-height: 100%;
  height: auto;
  width: 100%;
  margin: auto;
  z-index: 0;
`

const StyledHeading = styled(Heading)`
  position: relative;
  margin: 0;
  overflow: hidden;
  color: ${theme('colors.white')};
  text-align: center;
  text-overflow: ellipsis;
  text-shadow: 0 1px 0 rgba(0, 0, 0, 0.38);
  z-index: 2;

  ${breakpoint('m')`
    font-size: ${theme('fonts.size.xl')};
  `}
`

const ThumbnailPoster = ({ image, title }) => (
  <Wrapper>
    <StyledImage link={image} />
    <StyledHeading level={3}>{title}</StyledHeading>
  </Wrapper>
)

ThumbnailPoster.propTypes = {
  image: PropTypes.string,
  title: PropTypes.string,
}

export default ThumbnailPoster
