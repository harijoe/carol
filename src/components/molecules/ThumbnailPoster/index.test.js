import React from 'react'
import { shallow } from 'enzyme'

import ThumbnailPoster from './'

it('renders the ThumbnailPoster component', () => {
  expect(shallow(<ThumbnailPoster title="Test" image={{ src: 'test.jpg' }} />)).toMatchSnapshot()
})
