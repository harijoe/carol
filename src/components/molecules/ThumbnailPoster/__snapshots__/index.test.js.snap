exports[`test renders the ThumbnailPoster component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.figure>
    <Styled(Mock)
        link="Test image" />
    <Styled(Mock)
        level={3}>
        Test
    </Styled(Mock)>
</styled.figure>,
  "nodes": Array [
    <styled.figure>
      <Styled(Mock)
            link="Test image" />
      <Styled(Mock)
            level={3}>
            Test
      </Styled(Mock)>
</styled.figure>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ThumbnailPoster
        image="Test image"
        title="Test" />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "image": "Test image",
          "title": "Test",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.figure>
          <Styled(Mock)
                    link="Test image" />
          <Styled(Mock)
                    level={3}>
                    Test
          </Styled(Mock)>
</styled.figure>,
        "_debugID": 2,
        "_renderedOutput": <styled.figure>
          <Styled(Mock)
                    link="Test image" />
          <Styled(Mock)
                    level={3}>
                    Test
          </Styled(Mock)>
</styled.figure>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ThumbnailPoster
    image="Test image"
    title="Test" />,
}
`;
